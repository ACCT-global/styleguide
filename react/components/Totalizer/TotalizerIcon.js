import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

const CIRCLE_DIAMETER = '2.4rem'

class TotalizerIcon extends PureComponent {
  render() {
    const {
      item: { icon, iconBackgroundColor },
      testId,
    } = this.props

    return (
      <div
        data-testid={`${testId}__container`}
        className="flex items-center mr3"
      >
        <div
          data-testid={testId}
          className="flex items-center justify-center br-100 pa3 bg-light-silver c-muted-1"
          style={{
            height: CIRCLE_DIAMETER,
            width: CIRCLE_DIAMETER,
            backgroundColor: iconBackgroundColor,
          }}
        >
          {icon}
        </div>
      </div>
    )
  }
}

TotalizerIcon.propTypes = {
  item: PropTypes.shape({
    icon: PropTypes.node.isRequired,
    iconBackgroundColor: PropTypes.string,
  }).isRequired,
  testId: PropTypes.string,
}

export default TotalizerIcon
