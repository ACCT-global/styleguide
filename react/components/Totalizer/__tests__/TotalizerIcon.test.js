import React from 'react'
import { render } from 'react-testing-library'

import TotalizerIcon from '../TotalizerIcon'

describe('Totalizer/Icon spec', () => {
  const testId = 'testing'
  const bg = 'rgb(0, 0, 0)'
  const props = {
    item: {
      icon: <svg />,
      iconBackgroundColor: bg,
    },
    testId,
  }

  it('should match snapshot', () => {
    const Icon = render(<TotalizerIcon {...props} />)
    expect(Icon.asFragment()).toMatchSnapshot()
  })

  it('should render correctly', () => {
    const Icon = render(<TotalizerIcon testId={testId} {...props} />)
    const ElementToTest = Icon.getByTestId(testId)
    expect(ElementToTest).toBeTruthy()
    expect(ElementToTest.style.backgroundColor).toBe(bg)
    expect(ElementToTest.childElementCount).toBe(1)
  })
})
