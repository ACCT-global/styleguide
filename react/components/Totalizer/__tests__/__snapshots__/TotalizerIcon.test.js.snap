// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Totalizer/Icon spec should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center mr3"
    data-testid="testing__container"
  >
    <div
      class="flex items-center justify-center br-100 pa3 bg-light-silver c-muted-1"
      data-testid="testing"
      style="height: 2.4rem; width: 2.4rem; background-color: rgb(0, 0, 0);"
    >
      <svg />
    </div>
  </div>
</DocumentFragment>
`;
